src/lib/exhaust.ts

```diff
@@ -0,0 +1,31 @@
+import {ExhaustExportCsv} from '../models/exhaust-export-csv';
+import {ExhaustPluginInterface} from '../models/exhaust-plugin-interface';
+
+const createExhaustPlugins = (exhaustPluginConfigs: any) => {
+  return Object.keys(exhaustPluginConfigs).map((key: string) =>
+    createExhaustPlugin(key, exhaustPluginConfigs[key])
+  );
+};
+
+const createExhaustPlugin = (
+  pluginTypeName: string,
+  pluginConfigItems: any
+): ExhaustPluginInterface => {
+  switch (pluginTypeName) {
+    case 'csv':
+      return ExhaustExportCsv(pluginConfigItems);
+    default:
+      throw new Error(`unkonwn exhaust plugin type: ${pluginTypeName}`);
+  }
+};
+
+export const exhaust = (context: any, tree: any) => {
+  if (context && context.initialize.outputs) {
```

Comment:
you can check only for outputs, and receive outputs instead of context.
Since all other methods are parsing tree as a first argument, then options exhaust can follow the same rules